package/MDAnalysis/coordinates/GSD.py

```diff
@@ -111,20 +122,146 @@ def _read_frame(self, frame):
+    to be used to return to its original frame.
+
+    Parameters
+    ----------
+    file: :class:`gsd.fl.GSDFile`
+        File to access.
+
+    Example
+    -------
+    ::
+
+        gsdfileobj = gsd.fl.open(name=filename,
+                                     mode='rb',
+                                     application='gsd.hoomd '+gsd.__version__,
+                                     schema='hoomd',
+                                     schema_version=[1, 3]
+        file = GSDPicklable(gsdfileobj)
+        file_pickled = pickle.loads(pickle.dumps(file))
+
+    See Also
+    ---------
+    :func:`MDAnalysis.lib.picklable_file_io.FileIOPicklable`
+    :func:`MDAnalysis.lib.picklable_file_io.BufferIOPicklable`
+    :func:`MDAnalysis.lib.picklable_file_io.TextIOPicklable`
+    :func:`MDAnalysis.lib.picklable_file_io.GzipPicklable`
+    :func:`MDAnalysis.lib.picklable_file_io.BZ2Picklable`
+
+
+    .. versionadded:: 2.0.0
+    """
+    def __getstate__(self):
+        return self.file.name, self.file.mode
+
+    def __setstate__(self, args):
+        gsd_version = gsd.__version__
+        schema_version = [1, 4] if gsd_version >= '1.9.0' else [1, 3]
+        gsdfileobj = fl.open(name=args[0],
+                             mode=args[1],
+                             application='gsd.hoomd ' + gsd_version,
+                             schema='hoomd',
+                             schema_version=schema_version)
+        self.__init__(gsdfileobj)
+
+
+def gsd_pickle_open(name, mode='rb'):
+    """Open hoomd schema GSD file with pickle function implemented.
+
+    This function returns either GSDPicklable object. It can be used as a
+    context manager, and replace the built-in :func:`gsd.hoomd.open` function
+    in read mode that only returns an unpicklable file object.
+
+    Schema version will depend on the version of gsd module.
+
+    Note
+    ----
+    Can be only used with read mode.
+
+    Parameters
+    ----------
+    name : str
+        a filename given a text or byte string.
+    mode: {'r', 'rb'} (optional)
+        'r', 'rb':  open for reading;
+
+    Returns
+    -------
+    stream-like object: GSDPicklable
+
+    Raises
+    ------
+    ValueError
+        if `mode` is not one of the allowed read modes
+
+    Examples
+    -------
+    open as context manager::
+
+        with gsd_pickle_open('filename') as f:
+            line = f.readline()
+
+    open as function::
+
+        f = gsd_pickle_open('filename')
+        line = f.readline()
+        f.close()
+
+    See Also
+    --------
+    :func:`MDAnalysis.lib.util.anyopen`
+    :func:`MDAnalysis.lib.picklable_file_io.pickle_open`
+    :func:`MDAnalysis.lib.picklable_file_io.bz2_pickle_open`
+    :func:`MDAnalysis.lib.picklable_file_io.gzip_pickle_open`
+    :func:`gsd.hoomd.open`
+
+
+    .. versionadded:: 2.0.0
+    """
+    gsd_version = gsd.__version__
+    schema_version = [1, 4] if gsd_version >= '1.9.0' else [1, 3]
```

Comment:
Maybe something for me to do in a follow-up, but we can avoid some headaches long-term by using the PEP440 implementation of version comparisons here and elsewhere.

Those are robust to `alpha/beta/rc/dev` versions and so on. The raw string comparison is much less robust; there's a sort of middle ground available from distutils for version checking as well, though the PEP one is better these days.

Probably out of scope given the size of this PR already though.